tests/test_bgp_af.rb

```diff
@@ -519,103 +756,146 @@ def test_dampening
     #############################################
-    vrf = 'brown'
-    af = %w(ipv6 unicast)
+    if platform == :nexus
+      vrf = 'brown'
+      af = %w(ipv6 unicast)
+    elsif platform == :ios_xr
+      vrf = 'default'
+      af = %w(ipv4 unicast)
+      config('route-policy DropAllTraffic', 'end')
+    end
     bgp_af = RouterBgpAF.new(asn, vrf, af)
 
     bgp_af.dampening = 'DropAllTraffic'
 
-    pattern = /^ *dampening route-map DropAllTraffic$/
-
     # Check property got set
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    if platform == :nexus
+      pattern = /^ *dampening route-map DropAllTraffic$/
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+    elsif platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} " \
+                              'address-family ipv4 unicast bgp dampening')
+      pattern = /^ *bgp dampening route-policy DropAllTraffic$/
+    end
 
     assert_match(pattern, af_string,
                  "Error: 'dampening' is not configured and should be")
 
     # Check properties got set
-    af_string = get_bgp_af_dampening_params(asn, vrf, af)
-    pattern_params = 'Dampening policy configured: DropAllTraffic'
+    if platform == :nexus
+      af_string = get_bgp_af_dampening_params(asn, vrf, af)
+      pattern_params = 'Dampening policy configured: DropAllTraffic'
 
-    assert_match(pattern_params, af_string,
-                 'Error: dampening properties DropAllTraffic is not ' \
+      assert_match(pattern_params, af_string,
+                   'Error: dampening properties DropAllTraffic is not ' \
                    'configured and should be')
 
-    # Check getters
-    assert_equal(bgp_af.dampening, 'DropAllTraffic',
-                 'Error: dampening getter did not match')
-    assert_equal(bgp_af.dampening_routemap, 'DropAllTraffic',
-                 'Error: dampening getter did not match')
+      # Check getters
+      assert_equal(bgp_af.dampening, 'DropAllTraffic',
+                   'Error: dampening getter did not match')
+      assert_equal(bgp_af.dampening_routemap, 'DropAllTraffic',
+                   'Error: dampening getter did not match')
+    end
 
     #
     # Unset and verify
     #
     bgp_af.dampening = nil
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-    pattern = /^ *dampening$/
+
+    if platform == :nexus
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+      pattern = /^ *dampening$/
+    elsif platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} " \
+                              'address-family ipv4 unicast bgp dampening')
+      pattern = /^ *bgp dampening$/
+    end
 
     refute_match(pattern, af_string, "Error: 'dampening' is still configured")
 
+    config('no route-policy DropAllTraffic') if platform == :ios_xr
+
     #############################################
     # Set and verify 'dampening' with default   #
     #############################################
+    return if platform == :ios_xr
```

Comment:
OK. Split and added test_dampening_with_default test.

> On 30 Nov 2015, at 22:51, Glenn Matthews <notifications@github.com> wrote:
> 
> In tests/test_bgp_af.rb <https://github.com/cisco/cisco-network-node-utils/pull/87#discussion_r46209118>:
> 
> >      #############################################
> >      # Set and verify 'dampening' with default   #
> >      #############################################
> > +    return if platform == :ios_xr
> Instead of this, can you split the test case in half here and have the second test case report as skipped on XR?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/cisco/cisco-network-node-utils/pull/87/files#r46209118>.
>